### PR TITLE
rustdoc: Rename typedef to type alias

### PR DESCRIPTION
**Pull request by camelid**
This matches the name used by the [Rust Reference][1], which is also what
people usually call these items.

[1]: https://doc.rust-lang.org/reference/items/type-aliases.html

r? @GuillaumeGomez

### CODE REVIEW

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Thanks!

r=me once CI pass

File: src/tools/jsondoclint/src/item_kind.rs [hunk 1]
```diff
@@ -186,7 +186,7 @@ impl Kind {
             ItemKind::StructField => StructField,
             ItemKind::Trait => Trait,
             ItemKind::TraitAlias => TraitAlias,
-            ItemKind::Typedef => Typedef,
+            ItemKind::TypeAlias => Typedef,
```
Comment (aDotInTheVoid) [inline src/tools/jsondoclint/src/item_kind.rs:189]: NIT: Would you mind renaming `Typedef` here for consistency.

File: src/tools/jsondoclint/src/validator.rs [hunk 2]
```diff
@@ -221,7 +221,7 @@ impl<'a> Validator<'a> {
         }
     }
 
-    fn check_typedef(&mut self, x: &'a Typedef) {
+    fn check_typedef(&mut self, x: &'a TypeAlias) {
```
Comment (aDotInTheVoid) [inline src/tools/jsondoclint/src/validator.rs:224]: NIT: Could you rename this to `check_typealias`

**Review by aDotInTheVoid — CHANGES_REQUESTED** [review]
Looks good WRT Rustdoc-JSON couple of naming nits to keeps the tools consistant with the lib. Running a find and replace of `inner.typedef` to `inner.type_alias` should fix most of the rustdoc-json tests.

File: src/tools/jsondoclint/src/item_kind.rs (hunk 1 above)
Comment (camelid) [reply src/tools/jsondoclint/src/item_kind.rs:189]: Sure, not sure why I didn't the first time around 😅

File: src/tools/jsondoclint/src/validator.rs (hunk 2 above)
Comment (camelid) [reply src/tools/jsondoclint/src/validator.rs:224]: Ditto, thanks.

**Comment by jdahlstrom** [comment]
Just wanted to say thanks :D The use of a wrong term bothered me a while ago but didn't create an issue back then. Now I thought I'd finally create one, but when I checked the docs it was already fixed 👍

**Pull request merged**